### PR TITLE
Remove the need for sprite icon images; update Icons file

### PR DESCRIPTION
**Pull request by VoxelDoesCode**
It felt like that it was very unoptimized to have two images for the demo buttons. Initially, I wanted to merge it into one image, but it was a much better choice to have it so that we use the Icons font file for the sprites instead. Turns out, considering how the typeface is rendered, it looks much smoother and crisper. The contrast between the squares and the symbols is much clearer alongside!

Before the change:
unknown
After the change:
unknown

Note that there is a few compromises that had to be made with the limited symbol choices. As of publishing this there is not a symbol for a crossed out keyboard, but I hope the eye works just fine for now.

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus_demo.cpp [hunk 1]
```diff
@@ -34,17 +34,17 @@ int CMenus::DoButton_DemoPlayer(const void *pID, const char *pText, int Checked,
 	return UI()->DoButtonLogic(pID, Checked, pRect);
 }
 
-int CMenus::DoButton_Sprite(CButtonContainer *pButtonContainer, int ImageID, int SpriteID, int Checked, const CUIRect *pRect, int Corners)
-{
+int CMenus::DoButton_Sprite(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners)
```
Comment (Robyt3) [inline src/game/client/components/menus_demo.cpp:37]: Rename the method to `DoButton_FontIcon`, as it's no longer doing anything related to sprites and instead draws a font icon now.

**Comment by C0D3D3V** [comment]
You can look into the code how we made the verified symbol for servers. Ao if you want to you could overlap a crossed out symbol with the keyvoard. 
Also does the save symbol makes more sense then the camera symbol? 

I know we have done the same in the Server browser. But we should remember that this will make customization a lot harder.

**Comment by Robyt3** [comment]
You could also replace the `Close` button with an `X` icon like in the main menu.

And make the margin between the right buttons smaller so it's the same size as the margin between the left buttons:

screenshot

File: src/game/client/components/menus_demo.cpp [hunk 2]
```diff
@@ -34,17 +34,17 @@ int CMenus::DoButton_DemoPlayer(const void *pID, const char *pText, int Checked,
 	return UI()->DoButtonLogic(pID, Checked, pRect);
 }
 
-int CMenus::DoButton_Sprite(CButtonContainer *pButtonContainer, int ImageID, int SpriteID, int Checked, const CUIRect *pRect, int Corners)
-{
+int CMenus::DoButton_FontIcon(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners)
+{
```
Comment (Chairn) [inline src/game/client/components/menus_demo.cpp:38]: trailing spaces

File: src/game/client/components/menus_demo.cpp [hunk 3]
```diff
@@ -34,17 +34,17 @@ int CMenus::DoButton_DemoPlayer(const void *pID, const char *pText, int Checked,
...
-	Graphics()->QuadsDrawTL(&QuadItem, 1);
-	Graphics()->QuadsEnd();
+
+	TextRender()->SetCurFont(TextRender()->GetFont(TEXT_FONT_ICON_FONT));
+	CUIRect Rect = *pRect;
+	CUIRect Temp;
+	Rect.HMargin(2.0f, &Temp);
+	SLabelProperties Props;
+	UI()->DoLabel(&Temp, pText, Temp.h * CUI::ms_FontmodHeight, TEXTALIGN_CENTER, Props);
+	TextRender()->SetCurFont(NULL);
```
Comment (Chairn) [inline src/game/client/components/menus_demo.cpp:47]: nullptr

File: src/game/client/components/menus.cpp [hunk 4]
```diff
@@ -2075,14 +2075,14 @@ int CMenus::Render()
...
-			if(DoButton_Sprite(&s_SlowDownButton, IMAGE_DEMOBUTTONS, SPRITE_DEMOBUTTON_SLOWER, 0, &Button, IGraphics::CORNER_ALL))
+			if(DoButton_FontIcon(&s_SlowDownButton, "\xEF\x81\x8A", 0, &Button, IGraphics::CORNER_ALL))
 				DecDemoSpeed = true;
 
 			// fastforward
 			Part.VSplitLeft(5.0f, 0, &Part);
 			Part.VSplitLeft(ButtonSize, &Button, &Part);
 			static CButtonContainer s_FastForwardButton;
-			if(DoButton_Sprite(&s_FastForwardButton, IMAGE_DEMOBUTTONS, SPRITE_DEMOBUTTON_FASTER, 0, &Button, IGraphics::CORNER_ALL))
+			if(DoButton_FontIcon(&s_FastForwardButton, "\xEF\x81\x8e", 0, &Button, IGraphics::CORNER_ALL))
```
Comment (Chairn) [inline src/game/client/components/menus.cpp:2085]: caps 8E instead of 8e for consistency

File: src/game/client/components/menus_demo.cpp [hunk 5]
```diff
@@ -420,14 +420,14 @@ void CMenus::RenderDemoPlayer(CUIRect MainView)
...
-	if(DoButton_Sprite(&s_SlowDownButton, IMAGE_DEMOBUTTONS, SPRITE_DEMOBUTTON_SLOWER, 0, &Button, IGraphics::CORNER_ALL))
+	if(DoButton_FontIcon(&s_SlowDownButton, "\xEF\x81\x8A", 0, &Button, IGraphics::CORNER_ALL))
 		DecreaseDemoSpeed = true;
 
 	// fastforward
 	ButtonBar.VSplitLeft(Margins, 0, &ButtonBar);
 	ButtonBar.VSplitLeft(ButtonbarHeight, &Button, &ButtonBar);
 	static CButtonContainer s_FastForwardButton;
-	if(DoButton_Sprite(&s_FastForwardButton, IMAGE_DEMOBUTTONS, SPRITE_DEMOBUTTON_FASTER, 0, &Button, IGraphics::CORNER_ALL))
+	if(DoButton_FontIcon(&s_FastForwardButton, "\xEF\x81\x8e", 0, &Button, IGraphics::CORNER_ALL))
```
Comment (Chairn) [inline src/game/client/components/menus_demo.cpp:430]: caps 8E

**Comment by Chairn** [comment]
Too bad, that camera icon was awesome :D
It's kinda weird that the cut icon arrows both point right, no?

**Comment by VoxelDoesCode** [comment]
> But we should remember that this will make customization a lot harder.
I don't know anyone who customizes these icons. nor do I know anyone who would.

> It's kinda weird that the cut icon arrows both point right, no?
Well, It's closest we have to any cut indicator, unless we rewrite the system. Also, it's possible to change it to a camera, I just have to find the right icon for it.

File: src/game/client/components/menus.h [hunk 6]
```diff
@@ -78,7 +78,7 @@ class CMenus : public CComponent
 	char m_aLocalStringHelper[1024];
 
 	int DoButton_DemoPlayer(const void *pID, const char *pText, int Checked, const CUIRect *pRect);
-	int DoButton_FontIcon(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners);
+	int DoButton_FontIcon(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners, bool IsDisabled);
```
Comment (Chairn) [inline src/game/client/components/menus.h:81]: = false here, not in the .cpp :)

**Comment by VoxelDoesCode** [comment]
Do you think it'd be possible to replace file_icons.png? Or would be removing too much?

**Comment by Chairn** [comment]
I think it's fine if you find close enough characters

File: src/game/client/components/menus.h [hunk 7]
```diff
@@ -78,7 +78,7 @@ class CMenus : public CComponent
 	char m_aLocalStringHelper[1024];
 
 	int DoButton_DemoPlayer(const void *pID, const char *pText, int Checked, const CUIRect *pRect);
-	int DoButton_Sprite(CButtonContainer *pButtonContainer, int ImageID, int SpriteID, int Checked, const CUIRect *pRect, int Corners);
+	int DoButton_FontIcon(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners, bool IsDisabled);
```
Comment (Robyt3) [inline src/game/client/components/menus.h:81]:
```suggestion
	int DoButton_FontIcon(CButtonContainer *pButtonContainer, const char *pText, int Checked, const CUIRect *pRect, int Corners, bool Enabled = true);
```

Use the positive term and the inverse bool literals.

Comment (VoxelDoesCode) [reply src/game/client/components/menus.h:81]: Would I have to set the if statement to !Enabled, or would I have to changed all of the functions to false?

**Comment by VoxelDoesCode** [comment]
When testing this, I did get some lag spikes when un-focusing from the client while selecting a file, and then going back in and trying to hit cancel. It freezes for a split second and doesn't receive the input. I have no idea what could be causing that, but it's something to bring up nonetheless.

**Pull request merged**